Automatically build and push docker image to docker.io/dashpay/dashd-develop

codablock opened the pull request:
This builds and pushes a simplified docker image to dashpay/dashd-develop.
I plan to use this image to automatically deploy devnets via Ansible.

This conflicts with what we have in dashevo/dash already. I will update the dashevo version to this as well later. The reason I chose to create a bare/simplified version instead of what we had before is that we should continue developing the official Docker Images in https://github.com/dashpay/docker-dashd instead of duplicating stuff here. As the new image is only meant to be used from automatic deployments, it doesn't make much sense to have any default configs or helper scripts to bootstrap configs.

nmarley commented on docker/Dockerfile:
```diff
@@ -0,0 +1,7 @@
+FROM debian:stretch
+LABEL maintainer="Dash Evolution Developers <evodevs@dash.org>"
```
I have been changing this to "Dash Developers <dev@dash.org>" in order to reflect our unified team/common goal, but not super critical.

codablock replied:
Updated label

UdjinM6 commented on .gitignore:
```diff
@@ -132,3 +132,5 @@ dash-cli
 dash-qt
 make
+
+/docker/bin
```
Bring new line at eof back pls

codablock commented:
Fixed the new-line in .gitignore and also fixed a problem pushing the image (as seen in the pr_docker branch, not in the PR). @UdjinM6 Can you also re-push the pr_docker branch?

The build errors really look like internal problems.

UdjinM6 commented:
Hmm... still fails 🤔

codablock commented:
I added a few lines in .travis.yml to print out some system resources. The errors happen while print(stdout) is called, and as rpc-tests.py has been changed recently to write logs to a temporary file, it could be that /tmp is out of space.

UdjinM6 commented:
Weird. Pushed updated `pr_docker`, let's see.

codablock commented:
The issue we see here seems to be https://github.com/travis-ci/travis-ci/issues/4704. It behaves differently in the PR and in the branch build because secrets are only enabled on branch builds. The usual recommendation to set "filter_secrets: false" is not an option for us as I'd assume, as we are actually introduced the use of secrets with this PR.

I added a fix that found here: https://github.com/firehol/netdata/pull/3183
If that doesn't work, I'll look for another workaround.

codablock commented:
Build is finally gree: https://travis-ci.org/dashpay/dash/builds/322244635?utm_source=github_status&utm_medium=notification

I rebased and reworded the fix for the Travis failure.

codablock commented:
Rebased on develop and pushed a fix to make it work with develop.
Also removed a bogus example from .travis.yml and switched the branch on which we push to "develop".

Should be ready for merge now.

codablock commented:
Now travis is timing out on the wine tests, but this time because the total run-time of the job exceeds 50 minutes.

I checked the Bitcoin code and they don't run the wine tests at all. What was the reason for enabling them? Maybe it's time to disable them?

UdjinM6 commented:
IIRC there were some edge cases when dash-qt crashed in wine which were caught only by tests running in wine specifically. The main reason for slow tests in wine is performance of the boost, see https://github.com/bitcoin/bitcoin/pull/8650. Backporting 8650 (https://github.com/UdjinM6/dash/commit/d7791d9b206d91799b43e4f7a101d72ff71e6905) helps to speed things up significantly and lower the time of the test down to ~4 minutes https://travis-ci.org/UdjinM6/dash/jobs/323291048#L5708

codablock commented:
https://github.com/bitcoin/bitcoin/pull/8650 seems to never have been merged as it turned out that it would remove some functionality and https://github.com/bitcoin/bitcoin/pull/8671 would be enoug. https://github.com/bitcoin/bitcoin/pull/8671 is already merged into Dash, but it seems not to help. I'd prefer to not backport 8650 as it would result in many merge conflicts and too much diverging code.

Also, it looks like we are running wine tests for Win32 and Win64, while only Win32 has the problem of long running tests. The Win64 tests only need 189 seconds as of https://travis-ci.org/dashpay/dash/jobs/322913738#L2507.
What about disabling Win32 tests for now and only run Win64 tests? Would this still catch the problems from the past?
